spacemacs/local/hybrid-mode/hybrid-mode.el

```diff
@@ -56,27 +93,31 @@
       (hybrid-mode-setup-keymaps)
     (hybrid-mode-restore-keymaps)))
 
+(defun hybrid-mode-activate-keymap ()
+  (setq hybrid-mode-insert-state-map-active t))
+
+(defun hybrid-mode-deactivate-keymap ()
+  (setq hybrid-mode-insert-state-map-active nil))
+
 (defun hybrid-mode-setup-keymaps ()
   "Enter the church of Emacs in insert mode only."
+  (add-hook 'evil-insert-state-entry-hook 'hybrid-mode-activate-keymap)
+  (add-hook 'evil-insert-state-exit-hook 'hybrid-mode-deactivate-keymap)
+  (add-to-list 'minor-mode-map-alist
+               `(hybrid-mode-insert-state-map-active . ,hybrid-mode-insert-state-map))
   (setq hybrid-mode-insert-state-map-backup evil-insert-state-map
-        hybrid-mode-insert-state-cursor-backup evil-insert-state-cursor)
-  (setq evil-insert-state-cursor (list (car evil-emacs-state-cursor)
-                                       (cadr evil-insert-state-cursor)))
-  (copy-face 'spacemacs-insert-face 'hybrid-mode-insert-face-backup)
-  (copy-face 'spacemacs-emacs-face 'spacemacs-insert-face)
-  ;; (setcdr evil-insert-state-map nil)
-  ;; (define-key evil-insert-state-map [escape] 'evil-normal-state)
-  (dolist (key hybrid-mode-evil-insert-state-keys)
-    (define-key evil-insert-state-map key nil))
-  (define-key evil-insert-state-map
-    (read-kbd-macro evil-toggle-key) nil)
-  ;; see if the above works withou this
-  ;; (when (bound-and-true-p evil-escape-mode)
-  ;;   (define-key evil-insert-state-map "f" 'evil-escape-insert-state))
-  )
+        hybrid-mode-insert-state-cursor-backup evil-insert-state-cursor
+        evil-insert-state-cursor hybrid-mode-insert-state-cursor)
+  (copy-face 'hybrid-mode-insert-face 'spacemacs-insert-face)
+  (dolist (key-bnd hybrid-mode-default-evil-insert-state-bindings)
```

Comment:
Can you add a comment to explain what is done by the dolist ? Although it is easy to understand the "what is done" it is not so obvious to deduce the "why" which is to remove _non customized keys_ from the insert state map.